gridstatus/spp.py

```diff
@@ -355,6 +354,78 @@ def get_load_forecast_mid_term(
 
         return df
 
+    def _handle_dst_floor_date(
+        self,
+        date: pd.Timestamp,
+        freq: str = "5min",
+    ) -> pd.Timestamp:
+        """Handle DST transition when flooring a date.
+
+        Args:
+            date: The date to floor
+            freq: The frequency to floor to (e.g., "5min", "h")
+
+        Returns:
+            Timestamp floored to the specified frequency
+        """
+        try:
+            floored_date = date.floor(freq)
+        except pytz.AmbiguousTimeError:
+            floored_date = self.safe_for_dst_transition_floor(date, freq)
+
+        return floored_date
+
+    def _get_short_term_forecast_data(
+        self,
+        date: str | pd.Timestamp,
+        base_url: str,
+        file_prefix: str,
+        buffer_minutes: int = 2,
+    ) -> tuple[pd.DataFrame, str] | None:
+        """Get short-term forecast data with common DST handling logic.
+
+        Args:
+            date: Date to get data for. Supports "latest" and "today"
+            base_url: Base URL for downloads
+            file_prefix: Prefix for the file name (e.g., "OP-STLF", "OP-STRF")
+            buffer_minutes: Buffer minutes for "latest" date
+
+        Returns:
+            tuple: (dataframe, url) or None if date is in the future
+        """
+        if date == "latest":
+            date = self.now() - pd.Timedelta(minutes=buffer_minutes)
+
+        # Files do not exist in the future
+        if date > self.now():
+            return None
+
+        floored_date = self._handle_dst_floor_date(date, "5min")
+
+        hour = floored_date.hour
+        padded_hour = str(hour).zfill(2)
+        padded_hour_plus_one = str((hour + 1) % 24).zfill(2)
+
+        # NOTE: this needs to be updated for DST every year
+        # 0105d through 0155d have a "d" on 2025-11-02
+        add_d = (
+            floored_date.year == 2025
+            and floored_date.month == 11
+            and floored_date.day == 2
+            and floored_date.hour == 1
+            and floored_date.minute != 0
+        )
```

Comment:
I updated this to just use a hardcoded date to reduce complexity. It does mean we have to add to this logic next year but as discussed, I think we are fine with this for some datasets.